com.sap.cloud.lm.sl.cf.persistence/src/main/java/com/sap/cloud/lm/sl/cf/persistence/services/ObjectStoreFileStorage.java

```diff
@@ -65,19 +64,18 @@ public void addFile(FileEntry fileEntry, File file) throws FileStorageException
 
     @Override
     public List<FileEntry> getFileEntriesWithoutContent(List<FileEntry> fileEntries) throws FileStorageException {
-        Set<String> existingFiles = blobStore.list(container)
-            .stream()
-            .map(StorageMetadata::getName)
-            .collect(Collectors.toSet());
-
         return fileEntries.stream()
-            .filter(fileEntry -> {
-                String id = fileEntry.getId();
-                return !existingFiles.contains(id);
-            })
+            .filter(this::isFileNotPresent)

```

Comment:
This will now make the same request to the blob store for each file entry. This will (probably) degrade the performance by a lot.

Reply:
I will revert this change.